core/field_note.js

```diff
@@ -42,809 +35,797 @@ goog.require('goog.userAgent');
-  var whiteKeyGroup = Blockly.utils.createSvgElement('g', {}, this.pianoSVG_);
-  var blackKeyGroup = Blockly.utils.createSvgElement('g', {}, this.pianoSVG_);
-
-  // Add three piano octaves, so we can animate moving up or down an octave.
-  // Only the middle octave gets bound to events.
-  this.keySVGs_ = [];
-  this.addPianoOctave_(-this.fieldEditorWidth_ + Blockly.FieldNote.EDGE_PADDING,
-      whiteKeyGroup, blackKeyGroup, null);
-  this.addPianoOctave_(0, whiteKeyGroup, blackKeyGroup, this.keySVGs_);
-  this.addPianoOctave_(this.fieldEditorWidth_ - Blockly.FieldNote.EDGE_PADDING,
-      whiteKeyGroup, blackKeyGroup, null);
-
-  // Note name indicator at the top of the field
-  this.noteNameText_ = Blockly.utils.createSvgElement('text',
-      {
-        'x': this.fieldEditorWidth_ / 2,
-        'y': Blockly.FieldNote.TOP_MENU_HEIGHT / 2,
-        'class': 'blocklyText',
-        'text-anchor': 'middle',
-        'dominant-baseline': 'middle',
-      }, svg);
-
-  // Note names on the low and high C keys
-  var lowCX = Blockly.FieldNote.WHITE_KEY_WIDTH / 2;
-  this.lowCText_ = this.addCKeyLabel_(lowCX, svg);
-  var highCX = lowCX + (Blockly.FieldNote.WHITE_KEY_WIDTH *
-    (Blockly.FieldNote.NUM_WHITE_KEYS - 1));
-  this.highCText_ = this.addCKeyLabel_(highCX, svg);
-
-  // Horizontal line at the top of the keys
-  Blockly.utils.createSvgElement('line',
-      {
-        'stroke': this.sourceBlock_.getColourTertiary(),
-        'x1': 0,
-        'y1': Blockly.FieldNote.TOP_MENU_HEIGHT,
-        'x2': this.fieldEditorWidth_,
-        'y2': Blockly.FieldNote.TOP_MENU_HEIGHT
-      }, svg);
-
-  // Drop shadow at the top of the keys
-  Blockly.utils.createSvgElement('rect',
-      {
-        'x': 0,
-        'y': Blockly.FieldNote.TOP_MENU_HEIGHT,
-        'width': this.fieldEditorWidth_,
-        'height': Blockly.FieldNote.SHADOW_HEIGHT,
-        'fill': Blockly.FieldNote.SHADOW_COLOR,
-        'fill-opacity': Blockly.FieldNote.SHADOW_OPACITY
-      }, svg);
-
-  // Octave buttons
-  this.octaveDownButton = this.addOctaveButton_(0, true, svg);
-  this.octaveUpButton = this.addOctaveButton_(
-      (this.fieldEditorWidth_ + Blockly.FieldNote.INSET * 2) -
-      Blockly.FieldNote.OCTAVE_BUTTON_SIZE, false, svg);
-
-  this.octaveDownMouseDownWrapper_ =
-    Blockly.bindEvent_(this.octaveDownButton, 'mousedown', this, function() {
-      this.changeOctaveBy_(-1);
-    });
-  this.octaveUpMouseDownWrapper_ =
-      Blockly.bindEvent_(this.octaveUpButton, 'mousedown', this,function() {
-        this.changeOctaveBy_(1);
+  /**
+   * Show a field with piano keys.
+   * @private
+   */
+  showEditor_(event) {
+    // Mobile browsers have issues with in-line textareas (focus & keyboards).
+    super.showEditor_(event, this.useTouchInteraction_);
+
+    // If there is an existing drop-down someone else owns, hide it immediately and clear it.
+    Blockly.DropDownDiv.hideWithoutAnimation();
+    Blockly.DropDownDiv.clearContent();
```

Comment:
I don't think you actually need to do this. Could you try deleting it?

Reply:
Done